dali/driver/hid.py

```diff
@@ -714,40 +723,72 @@ def __init__(self, *args, **kwargs):
         self._response_available = asyncio.Event()
         self._response = None
 
+    def _initialise_device(self):
+        self._sn+=1
+        if self._sn > 255:
+            self._sn = 1
+        # Read firmware version; pick up the reply in _handle_read
+        data = struct.pack('BBBBBBBBBB', 0xAA, self._HASSEB_READ_FIRMWARE_VERSION,
+                            self._sn, 0, 0, 0, 0, 0, 0, 0)
+        os.write(self._f, data)
+      
+    def _construct(self, command):
+        # sequence number
+        self._sn+=1
+        if self._sn > 255:
+            self._sn = 1
+        frame_length = 16
+        if command.is_query:
+            expect_reply = 1
+        else:
+            expect_reply = 0
+        transmitter_settling_time = 0
+        if command.sendtwice:
+            send_twice = 10 # 10 ms delay between messages
+        else:
+            send_twice = 0
+        frame = command.frame.as_byte_sequence
+        byte_a, byte_b = frame
+        data = struct.pack('BBBBBBBBBB', 0xAA, self._HASSEB_DALI_FRAME, self._sn,
+                           frame_length, expect_reply,
+                           transmitter_settling_time, send_twice,
```

Comment:
how is send_twice actually sent twice !?

Reply:
It needs further testing to confirm, however in my testing the device sends it when send_twice is set > 0. value of send twice is used as delay in ms between the two sends.

Reply:
you are creating a single packet, with a format which I haven't seen documented, can you link where you saw this struct ?